### DISCUSSION TITLE
New terminal window remains blank and no prompt shows up

### DISCUSSION THREAD

**Discussion by dpo** [wezterm/wezterm#3527] [General]
This is the second time this happens to me now. I'm happily using Wezterm on macOS 12.6.3. At times, I try to open a new terminal window, and it simply remains blank. No prompt ever comes up, even though things were working before and I did not update the configuration.

Enabling the debug overlays shows this:
```console
wezterm version: 20230326-111934-3666303c aarch64-apple-darwin
Window Environment: macOS 12.6.3 (21G419)
OpenGL version: Apple M1 Max 4.1 Metal - 76.3
Enter lua statements or expressions and hit Enter.
Press ESC or CTRL-D to exit
15:23:33.496 INFO wezterm_gui::update > latest release 20230408-112425-69ae8472 is newer than current build 20230326-111934-3666303c
06:42:14.312 INFO wezterm_gui::update > latest release 20230408-112425-69ae8472 is newer than current build 20230326-111934-3666303c
07:08:01.173 INFO wezterm_gui::update > latest release 20230408-112425-69ae8472 is newer than current build 20230326-111934-3666303c
23:49:31.951 WARN wezterm_font > No fonts contain glyphs for these codepoints: \u{2095}.
Placeholder glyphs are being displayed instead.
You may wish to install additional fonts, or adjust your
configuration so that it can find them.
https://wezfurlong.org/wezterm/config/fonts.html has more information about configuring fonts.
Set warn_about_missing_glyphs=false to suppress this message.
07:15:59.877 INFO wezterm_gui::update > latest release 20230408-112425-69ae8472 is newer than current build 20230326-111934-3666303c
16:42:01.454 ERROR wezterm_gui::termwindow::render > paint_opengl_pass failed: No such file or directory (os error 2)
>
```

Here is the `wezterm-gui-log`: https://gist.github.com/dpo/3fb74b0dd8d618eb4b2eb4998c302392

I'm not sure if this is relevant, but both times this happened, a notice that a Wezterm update is available came up at some point when I opened a new tab. I typically don't update right away because I have quite a bit of ongoing work.

Last time, I updated Wezterm, and it resolved the issue. However, I don't see why not updating should render the terminal unusable.

Any ideas?

**Comment by wez** [comment]
`paint_opengl_pass failed: No such file or directory (os error 2)` suggests that there was an error loading something from a file.
Most likely: you have a background image or something referenced in your config that is (no longer?) present on the local disk.

Hard to do more to help with the limited information provided; if you open an issue using the bug report template and include your configuration that might help to shed more light.

**Comment by dpo** [comment]
Here's my config:
```lua
local wezterm = require 'wezterm'

-- from https://wezfurlong.org/wezterm/config/lua/wezterm.gui/get_appearance.html
function scheme_for_appearance(appearance)
  if appearance:find 'Dark' then
    -- return 'Builtin Solarized Dark'
    -- return "3024 Night"
    return "OneHalfDark"
  else
    -- return 'Builtin Solarized Light'
    -- return "3024 Day"
    return "OneHalfLight"
  end
end

return {
  font = wezterm.font 'JetBrains Mono',
  color_scheme = scheme_for_appearance(wezterm.gui.get_appearance()),
  hide_tab_bar_if_only_one_tab = true,
  send_composed_key_when_left_alt_is_pressed = true, -- for accentuated characters
}
```

I don't believe the config is the issue. I can open new tabs in existing windows. I just can't open a new window.

**Reply by wez** [reply]
Weird. If you press `CTRL-L` or `CTRL-SHIFT-K` (both clear the screen/scrollback, but in slightly different ways), do you start to see the text?

There's very little in wezterm that would try to access files without you explicitly having something in your config, and even fewer things that would try to do that while rending the screen.

**Reply by dpo** [reply]
I'll have to get back to you later if/when this issue comes up again. I had to reboot and the issue went away.

**Reply by dpo** [reply]
So it happens again. And yes, `CTRL-L` shows the text.

**Reply by wez** [reply]
Sounds like maybe a certain character is being displayed and there's a problem loading a font or something.
If you can figure out how to reproduce this, please file a bug report issue using the issue template:
https://github.com/wez/wezterm/issues/new?assignees=&labels=bug&template=bug.yml

**Reply by dpo** [reply]
I commented out everything in my `~/.zshrc`, `~/.zshenv`, `~/.zprofile`, etc., but the same behavior occurs. The fact that it doesn't happen when I reboot or update WezTerm, but only creeps in after a while suggests that the source of the issue isn't in my startup files though. Can nobody reproduce this?

**Reply by wez** [reply]
You are the only person to have reported anything like this!

**Reply by moecasts** [reply]
@wez I have this problem too.

<img width="606" alt="image" src="https://github.com/wez/wezterm/assets/37169906/69f354b1-2429-4430-a78e-9cc02c0a04f6">

**Reply by cannonrush** [reply]
@wez I have the same problem with a fresh macOS install. This happens anytime I try to open WezTerm on my second monitor.
In the logs I see a message below:
```
15:05:14.138  WARN   wezterm_gui::termwindow::resize > cannot resize window to match Some(RowsAndCols { rows: 24, cols: 80 }) because window_state is MAXIMIZED
```
Resizing the window resolves the issue.

**Reply by wez** [reply]
@nvinogr sounds like maybe you're experiencing a variation of https://github.com/wez/wezterm/issues/2958

**Comment by denppa** [comment]
@wez Here to report the same problem.

I did not load any extra files, and a new tab also appears blank until I clear scrollback/screen of the new tab. This only happens on the wezterm window that has many tabs that has many panes within those tabs with a lot of log scrollback. So perhaps it has to do with too much scrollback in those other tabs?

**Reply by wez** [reply]
@denppa maybe you're also experiencing https://github.com/wez/wezterm/issues/2958 or something similar?

**Reply by denppa** [reply]
I looked at the issue, but that doesn't seem to be the problem for me. For context, my window is full screen, and the output of `tput lines` is exact.(i.e. I pressed enter that many times to confirm) I assume the `tput cols` output is also accurate.

Now it also happens for other windows that do not have logs scrolling through it and even completely new windows. I will try to update to nightly when I can shut the terminal down and see if that helps at all.

Installed through `brew --cask` at the moment, will be tapping into the repo and install the nightly version later.

```
21:58:02.974 ERROR wezterm_gui::termwindow::render::paint > paint_pass failed: No such file or directory (os error 2)
```
Update:
Just installed nightly, the problem has yet to appear.

**Comment by alg** [comment]
Reporting the same. Here are the cases I observe:

I have several tabs with contents showing just fine. I create a new tab and get a blank pane. The background color is correct, nothing appears. 

On several occasions I had one pane blank and one pane fully working in a split window.

If I run something with TUI (for example, neomutt) in the blank pane the contents of the app appear. Running "ls" doesn't have the same effect. If I exit the app, it goes to the blank pane again.

Hope it helps.

**Reply by alg** [reply]
BTW, just tried running "clear". The pane now displays the contents correctly. Unfortunately it's not always possible to run it.

Another observation is when you create a tab that becomes blank, I still can see the top tabbar momentarily. Switching to the previously visible tab displays it, going back to the created (invisible) removes it.

**Reply by pimmee** [reply]
I'm having the exact same issue.

**Comment by LHAASOIHEP** [comment]
only zshrc loaded
zprofile not loaded

**Comment by quantonganh** [comment]
I'm not sure if this is related, but in my case, sometimes after closing a program (Helix, Lazygit, ...), no prompt appears, and I have to press Enter to make it show.

By running WezTerm in debug mode, I found out that `wezterm_gui::termwindow::render::paint` takes a few seconds:

```
11:18:38.199  DEBUG  wezterm_gui::termwindow::render::paint  > paint_impl before call_draw elapsed=515.084125ms
11:18:38.199  DEBUG  wezterm_gui::termwindow::render::paint  > paint_impl elapsed=515.338833ms, fps=2.208928
11:18:43.719  TRACE  mux                                     > read_pty pane 1 read 1024 bytes
```

```
11:55:25.950  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl before call_draw elapsed=919.5µs
11:55:25.950  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl elapsed=1.323708ms, fps=3.9035459
11:55:29.034  DEBUG  window::os::macos::window              > key_common: chars=`;` unmod=`;` modifiers=`NONE` virtual_key=41 key_is_down:true
```

```
12:21:44.635  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl before call_draw elapsed=728.75µs
12:21:44.636  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl elapsed=1.128292ms, fps=0.76410156
12:21:46.771  DEBUG  window::os::macos::window              > key_common: chars=`j` unmod=`j` modifiers=`NONE` virtual_key=38 key_is_down:true
```

```
12:31:26.018  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl before call_draw elapsed=2.484958ms
12:31:26.018  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl elapsed=2.74825ms, fps=0
12:31:26.939  TRACE  window                                 > Notification(Any { .. })
12:31:26.939  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
12:31:26.940  TRACE  window                                 > Notification(Any { .. })
12:31:26.940  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
12:31:26.940  TRACE  window                                 > Notification(Any { .. })
12:31:26.940  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
12:31:30.087  TRACE  window                                 > FocusChanged(true)
12:31:30.088  DEBUG  wezterm_gui::termwindow                > FocusChanged(true)
```

```
15:12:00.363  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl before call_draw elapsed=99.833µs
15:12:00.364  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl elapsed=358.375µs, fps=0
15:12:01.260  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
15:12:01.261  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
15:12:01.261  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
15:12:05.895  DEBUG  wezterm_gui::termwindow                > FocusChanged(true)
15:12:05.901  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
```

```
15:24:31.128  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl before call_draw elapsed=84.667µs
15:24:31.128  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl elapsed=325.5µs, fps=0
15:24:32.029  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
15:24:32.029  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
15:24:32.030  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
15:24:35.249  DEBUG  wezterm_gui::termwindow                > FocusChanged(true)
15:24:35.250  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
```

What could be the reason? Is it related to the Fish shell?
